jerry-core/ecma/base/ecma-helpers-values-collection.c

```diff
@@ -67,7 +68,14 @@ ecma_new_values_collection (const ecma_value_t values_buffer[], /**< ecma values
 
     JERRY_ASSERT (cur_value_buf_iter_p + 1 <= cur_value_buf_end_p);
 
-    *cur_value_buf_iter_p++ = ecma_copy_value (values_buffer[value_index], do_ref_if_object);
+    if (do_ref_if_object)
+    {
+      *cur_value_buf_iter_p++ = ecma_copy_value (values_buffer[value_index]);
+    }
+    else
+    {
+      *cur_value_buf_iter_p++ = ecma_copy_value_if_not_object (values_buffer[value_index]);
+    }
```

Comment:
Unfortunalty it is used with both. Four times with do_ref_if_object and four times without it.